### PR TITLE
Add fr.handbrake.ghb.Plugin.dvdcss

### PR DESCRIPTION
**Pull request by daviewales**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly. `libdvdcss` extension for Handbrake. (Enables encoding commercial DVDs.)
- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [ ] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

This is a resubmission of #1863.
I've updated a few things to pass linter checks, match the Gnome version used by Handbrake, use latest version of `libdvdcss`, etc.

Handbrake developers do not want to include `libdvdcss` in the main Flatpak as they do not support it. (See https://github.com/HandBrake/HandBrake/pull/3164 for a long discussion.)
I've tested, and this PR successfully enables the Handbrake Flatpak to decode commercial DVDs.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+    "app-id": "fr.handbrake.ghb.Plugin.dvdcss",
+    "branch": "1",
```
Comment (bbhtt) [inline fr.handbrake.ghb.Plugin.dvdcss.json:3]: do you need a branch? Unless there are api/abi breaks involved there is no need to have a branch.

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                "sources": [
+                    {
+                        "type": "file",
+                        "path": "fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml"
+                    }
+                ],
+                "buildsystem": "simple",
+                "build-commands": [
+                    "install -Dp -m 644 fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml --target-directory=/app/extensions/dvdcss/share/metainfo",
+                    "appstream-compose --basename=fr.handbrake.ghb.Plugin.dvdcss --prefix=/app/extensions/dvdcss/ --origin=flatpak fr.handbrake.ghb.Plugin.dvdcss"
```
Comment (bbhtt) [inline fr.handbrake.ghb.Plugin.dvdcss.json:25]:
```suggestion
                    "install -Dp -m 644 fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml --target-directory=/app/extensions/dvdcss/share/metainfo"
```

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+    "app-id": "fr.handbrake.ghb.Plugin.dvdcss",
+    "branch": "1",
+    "runtime": "fr.handbrake.ghb",
+    "runtime-version": "stable",
+    "sdk": "org.gnome.Sdk//46",
+    "build-extension": true,
+    "separate-locales": false,
+    "appstream-compose": false,
```
Comment (bbhtt) [inline fr.handbrake.ghb.Plugin.dvdcss.json:9]:
```suggestion

```

File: fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  <extends>fr.handbrake.ghb</extends>
+  <name>Commercial DVD support for HandBrake</name>
+  <summary>Provides commercial DVD support with libdvdcss</summary>
+  <description>
+    <p>
+    This plugin adds support for using commercial DVDs as encoding sources. It provides the libdvdcss library from VLC.
+    </p>
+    <p>
+    NOTE: This plugin is not verified by, affiliated with, or supported by the HandBrake developers.
+    </p>
```
Comment (bbhtt) [inline fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml:13]:
```suggestion
    <p>
    <em>NOTE</em>: This plugin is not verified by, affiliated with, or supported by the HandBrake developers.
    </p>
```

Also make this the first p tag

File: libdvdcss.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "libdvdcss",
```
Comment (bbhtt) [inline libdvdcss.json:2]: this should be in the main manifest

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    "build-options": {
+        "prefix": "/app/extensions/dvdcss"
+    },
+    "modules": [
```
Comment (bbhtt) [inline fr.handbrake.ghb.Plugin.dvdcss.json:13]: did this get auto sorted ? It should be module name -> buildsystem -> build-commands -> sources

File: libdvdcss.json (hunk 5 above)
Comment (bbhtt) [reply libdvdcss.json:2]: no I mean copy the module from here to the main manifest and delete this extra manifest

**Comment by daviewales** [comment]
I've applied the requested changes.

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    "separate-locales": false,
+    "build-options": {
+        "prefix": "/app/extensions/dvdcss"
+    },
+    "modules": [
+        {
+            "name": "metadata",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dp -m 644 fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml --target-directory=/app/extensions/dvdcss/share/metainfo"
```
Comment (bbhtt) [inline fr.handbrake.ghb.Plugin.dvdcss.json:16]:
```suggestion
                "install -Dp -m 644 fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml --target-directory=${FLATPAK_DEST}/share/metainfo"
```

Comment (daviewales) [reply fr.handbrake.ghb.Plugin.dvdcss.json]: Thanks! Done.

File: fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    This plugin adds support for using commercial DVDs as encoding sources. It provides the libdvdcss library from VLC.
+    </p>
+</description>
+  <url type="vcs-browser">https://code.videolan.org/videolan/libdvdcss</url>
+  <url type="homepage">https://videolan.videolan.me/libdvdcss/</url>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0-or-later</project_license>
+  <developer id="org.videolan">
+      <name>VideoLan</name>
+  </developer>
```
Comment (hfiguiere) [inline fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml:21]: it's missing `releases`

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml"
+                }
+            ]
+        },
+        {
+            "name": "libdvdcss",
+            "config-opts": [ "--enable-shared", "--disable-static" ],
```
Comment (hfiguiere) [inline fr.handbrake.ghb.Plugin.dvdcss.json:27]:
```suggestion
            "config-opts": [ 
                "--enable-shared", 
                "--disable-static" 
            ],
```

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    "runtime-version": "stable",
+    "sdk": "org.gnome.Sdk//46",
+    "build-extension": true,
+    "separate-locales": false,
+    "build-options": {
+        "prefix": "/app/extensions/dvdcss"
+    },
+    "modules": [
+        {
+            "name": "metadata",
```
Comment (hfiguiere) [inline fr.handbrake.ghb.Plugin.dvdcss.json:13]: this doesn't need to be a separate module

Comment (daviewales) [reply fr.handbrake.ghb.Plugin.dvdcss.json:13]: I've tried merging them into a single module, but I've hit some issues.

If I create a single module with the union of the sources lists, I have to decide which buildsystem to use.

By default, the `libdvdcss` module appears to automatically build with autotools as required.

When combining the two modules, if I remove the `buildsystem: simple` directive, then the `metainfo.xml` file fails to install, but the `libdvdcss` library builds and installs correctly.

But if I include `buildsystem: simple` and `build-commands`, then `metainfo.xml` installs correctly, but it no longer automatically builds `libdvdcss` with autotools.

I thought perhaps I could just manually add all the required build commands to the `build-commands` list. However, it's not clear to me that this is any better than having two modules.
Also, it seems to build in a slightly different way, and produces extra files which weren't included before.

Could you provide some direction here?

Comment (bbhtt) [reply fr.handbrake.ghb.Plugin.dvdcss.json:13]: merge the sources and add `install -Dm0644 ...` to `post-install` of the other module.

Comment (daviewales) [reply fr.handbrake.ghb.Plugin.dvdcss.json]: Thanks!

**Comment by daviewales** [comment]
Somewhere along the way I've broken this so that the extension builds, but doesn't seem to install. (There are no errors, it's just not available to the app.) I'll see if I can fix it.

**Comment by bbhtt** [comment]
> Adding back `"branch": "1"` seems to fix it. Thoughts?

Looks like you need it because the extension point has it, but I think it should be removed from there too unless API/ABI breaks are involved 

https://github.com/flathub/fr.handbrake.ghb/blob/fd4c270b80d5264d8ca5e122b219f60d83bb3e0a/fr.handbrake.ghb.json#L26

**Comment by daviewales** [comment]
Keep it for this PR then? If so I'll edit the commit and force push the branch.

**Comment by bbhtt** [comment]
Yea.

**Comment by daviewales** [comment]
Done.

**Comment by bbhtt** [comment]
bot, build fr.handbrake.ghb.Plugin.dvdcss

**Comment by daviewales** [comment]
Looks like the x86_64 build succeeded, but the aarch64 build failed because the main Handbrake package is not available to install. See also: https://github.com/HandBrake/HandBrake/issues/4225

**Comment by bbhtt** [comment]
the app is not available for aarch64, you don't need to build the extension for that https://github.com/flathub/fr.handbrake.ghb/blob/master/flathub.json

**Comment by daviewales** [comment]
Thanks. I think I've fixed that.

**Comment by daviewales** [comment]
Seems that specifying the arch in the module doesn't prevent the builder from attempting to build for other architectures. Trying again with `flathub.json`. Do I have bot powers?

**Comment by daviewales** [comment]
bot, build fr.handbrake.ghb.Plugin.dvdcss

File: fr.handbrake.ghb.Plugin.dvdcss.json [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+                "install -Dp -m 644 fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml --target-directory=${FLATPAK_DEST}/share/metainfo"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "fr.handbrake.ghb.Plugin.dvdcss.metainfo.xml"
+                },
+                {
+                    "type": "git",
+                    "url": "https://code.videolan.org/videolan/libdvdcss.git",
```
Comment (hfiguiere) [inline fr.handbrake.ghb.Plugin.dvdcss.json:29]: usually the actual source (this) is first.

Comment (daviewales) [reply fr.handbrake.ghb.Plugin.dvdcss.json:25]: Thanks. Done.

**Comment by daviewales** [comment]
bot, build fr.handbrake.ghb.Plugin.dvdcss

File: libdvdcss.json (hunk 5 above)
Comment (bbhtt) [reply libdvdcss.json:2]: @daviewales you have to remove this now, it's a leftover.

Comment (daviewales) [reply libdvdcss.json:2]: Could you clarify? Do you mean I should remove the `"name": "libdvdcss", ` line? Or something else?

Comment (hfiguiere) [reply libdvdcss.json:2]: removing this file.

Comment (daviewales) [reply libdvdcss.json]: Thanks. Done.

**Comment by daviewales** [comment]
bot, build fr.handbrake.ghb.Plugin.dvdcss

**Pull request closed without merging**